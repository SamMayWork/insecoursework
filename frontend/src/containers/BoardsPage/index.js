import React, {useState, useEffect} from 'react';
import {makeStyles, useTheme} from '@material-ui/styles';
import Navbar from '../Navbar';
import Sidebar from '../../components/Sidebar';

import TestModules from './data.js';

import Board from '../../components/Board';

import testData from '../../components/PostSearch/tests/testData.js';
import PostSearch from '../../components/PostSearch';

const BoardsPage = props => {
	const [openSidebar, setOpenSidebar] = useState(false);
	const [boards, setBoards] = useState([]);
	
	// NOTE: Load boards from DB, here
	const updateBoards = boards => {
		 fetch("/get?all=1")
      .then(res => res.json())
      .then(
        (result) => {
        	let boards = result;
          setBoards(boards);
        },
        // Note: it's important to handle errors here
        // instead of a catch() block so that we don't swallow
        // exceptions from actual bugs in components.
        (error) => {
        	/*
          this.setState({
            isLoaded: true,
            error
          });
          */
        }
      )
	}
	
	const handleSidebarOpen = () => {
		setOpenSidebar(true);
	};
	const handleSidebarClose = () => {
		setOpenSidebar(false);
	};
	useEffect(() => {updateBoards(boards)}, [boards]);
	return (
		<div>
			<Navbar
				title = "Boards"
				onSidebarOpen = {handleSidebarOpen}
			/>
			<Sidebar
				onClose = {handleSidebarClose}
				open = {openSidebar}
				variant = {'temporary'}
			/>
			{
			/*
			<PostSearch
				board = {testData.board}
				boards = {testData.boards}
			/>
			*/
			}
			{boards.slice(0, 20).map((board, i) => (
				<Board
					key = {i}
					title = {board.board_module}
					year = {board.board_year}
					views = {(i+1) * 10}
				/>
			))}
		</div>
	);
};

export default BoardsPage;
