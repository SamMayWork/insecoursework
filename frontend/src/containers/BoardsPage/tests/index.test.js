import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

import BoardsPage from '../index';

// import testData from './testData.js';

const testComponent = props => {
	return (
		<BoardsPage/>
	);
}

describe('<BoardsPage/>', () => {
	describe('Validation testing', () => {
		/*
			NOTE: This test ensures that the page is loading the data from the DB
						correctly as the snapshot will have the correct data loaded inside
		*/
		it('should render and match the snapshot', () => {
			renderer.render(React.createElement(testComponent));
			const renderedOutput = renderer.getRenderOutput();
		  expect(renderedOutput).toMatchSnapshot();
		});
	});
});
