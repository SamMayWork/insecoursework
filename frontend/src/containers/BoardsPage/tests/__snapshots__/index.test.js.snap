// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardsPage/> Validation testing should render and match the snapshot 1`] = `<BoardsPage />`;
